resolver/resolver.cc

```diff
@@ -2245,9 +2245,29 @@ class ResolveTypeMembersAndFieldsWalk {
 };
 
 class ResolveSignaturesWalk {
+public:
+    struct MethodSignature {
+        core::LocOffsets loc;
+        ParsedSig sig;
+        vector<bool> argsToKeep;
+    };
+    struct ResolveMethodSignaturesJob {
+        core::FileRef file;
+        core::SymbolRef owner;
+        ast::MethodDef *mdef;
+        InlinedVector<MethodSignature, 1> sigs;
```

Comment:
My next move will be to try seeing if separating overloaded sigs from non-overloaded sigs results in a speed improvement. I suspect it will, as non-overloaded sigs will not require a vector here (or a `vector<bool>` within it).